openedx/core/djangoapps/team_api/views.py

```diff
@@ -0,0 +1,407 @@
+
+            * description: A description of the team.
+
+            * country: Optionally specifies which country the team is associated with.
+
+            * language: Optionally specifies which language the team is associated with.
+
+            * membership: A list of the users that are members of the team. See membership endpoint for more detail.
+
+            For all text fields, clients rendering the values should take care
+            to HTML escape them to avoid script injections, as the data is
+            stored exactly as specified. The intention is that plain text is
+            supported, not HTML.
+
+            If the user is not logged in, a 403 error is returned.
+
+            If the specified team does not exist, a 404 error is returned.
+
+        **Response Values for PATCH**
+
+            Only staff can patch teams. If the specified team does not exist, a
+            404 error is returned.
+
+            If "application/merge-patch+json" is not the specified content type,
+            a 415 error is returned.
+
+            If the update could not be completed due to validation errors, this
+            method returns a 400 error with all error messages in the
+            "field_errors" field of the returned JSON.
+    """
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser, OAuth2AuthenticationAllowInactiveUser)
+    permission_classes = (permissions.IsAuthenticated, IsStaffOrReadOnly)
+    lookup_field = 'team_id'
+    serializer_class = CourseTeamSerializer
+    parser_classes = (MergePatchParser,)
+
+    def get_queryset(self):
+        """
+        Returns the queryset used to access the given team.
+        """
+        return CourseTeam.objects.all()
+
+
+class TopicListView(APIView):
+    """HTTP endpoint for retreiving a list of topics for a given course."""
+
+    authentication_classes = (SessionAuthenticationAllowInactiveUser,)
+    permission_classes = (permissions.IsAuthenticated,)
+
+    page_size = 10
+    # TODO determine the correct max page size
+    max_page_size = sys.maxint
+
+    def get(self, request):
+        """
+        GET /api/team/v0/topics/?course_id={course_id}
+        """
+        try:
+            course_id_string = request.QUERY_PARAMS.get('course_id', None)
+            if course_id_string is None:
+                return Response(status=status.HTTP_400_BAD_REQUEST)
+
+            course_id = CourseKey.from_string(course_id_string)
+
+            course_module = modulestore().get_course(course_id)
+            if course_module is None:  # course is None if not found
+                return Response(status=status.HTTP_404_NOT_FOUND)
+
+            if CourseEnrollment.get_enrollment(request.user, course_id) is None:
+                return Response({'detail': "user must be enrolled"}, status=status.HTTP_403_FORBIDDEN)
+
+            topics = [t for t in course_module.teams_topics if t['is_active']]
```

Comment:
It's listed in the API spec, saying that topics which are not active should not be listed in the result of this call.